packages/ds-error-context/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use std::path::PathBuf;
+
+use derive_more::Display;
+
+#[derive(Display)]
```

Comment:
Oh interesting! I had been planning on checking out [`thiserror`](https://crates.io/crates/thiserror) which seems to have a similar API, but I guess we won't need that if we lean towards attaching `context()` to errors instead?

Reply:
So... we kinda use both. `thiserror` is used for programmatic errors that people can check for, handle, etc. The thing about this "error" enum is that it's really just an enum of strings that we can use to add context for end-users to reference. There can be an M:N behavior between actual errors and error contexts, for example. They're distinct entities.